### PR TITLE
Error validation

### PR DESCRIPTION
**Pull request by Fv40**
All changes to files other than test/MediatorClassTest.java rely on this PR: [https://github.com/EDOLK/CPS353-Project/pull/35](url), and will be resolved when it is merged to main

### CODE REVIEW

File: src/ComputeEngineImplementation.java [hunk 1]
```diff
@@ -25,17 +25,25 @@ public void setComputeEngineComputation(ComputeEngineComputation computeEngineCo
 
   @Override
   public EngineResponse submitRequest(UserRequest userRequest) {
+	try {
+		return submitRequestHelper(userRequest);
+	} catch (Exception e) {
```
Comment (kyliedelamater) [inline src/ComputeEngineImplementation.java:30]: maybe change this to IllegalArgumentException e. its probably good to be more specific

File: src/Mediator.java [hunk 2]
```diff
@@ -40,6 +40,15 @@ public <T> void sendInput(UserRequestProvider<T> provider) {
 		if(provider == null) {
 			throw new IllegalArgumentException("UserRequestProvider cannot be null");
 		}
+		try {
+			provider.propigateResponse(sendInputHelper(provider));
+		} catch (Exception e) {
```
Comment (kyliedelamater) [inline src/Mediator.java:45]: this as well you could do IOException e since its for input. at least i think it should work.

**Review by kyliedelamater — APPROVED** [review]
looks good, i would say just be a bit more specific about the type of error that would be thrown

File: src/ComputeEngineImplementation.java (hunk 1 above)
Comment (Fv40) [reply src/ComputeEngineImplementation.java:30]: So this change, and all other non-test changes, are from this branch: [https://github.com/EDOLK/CPS353-Project/pull/35](url), I merged it into mine to be able to create the integration test. That other pr hasn't been merged yet, so its changes will show up here until then.

**Pull request merged**